src/main/java/org/broadinstitute/consent/http/resources/DataSetResource.java

```diff
@@ -110,6 +111,45 @@ public Response createDataset(@Auth AuthUser user, @Context UriInfo info, String
         return Response.created(uri).entity(createdDataset).build();
     }
 
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/{datasetId}")
+    @PermitAll
+    public Response updateDataset(@Auth AuthUser user, @Context UriInfo info, String json, @PathParam("datasetId") Integer datasetId) {
```

Comment:
Super minor nit, I like to have all of the annotated arguments toward the front of the method, non-annotated to the end. I don't know if there's any style guide for this, but it's easier for me to read that way. I welcome other opinions.
```suggestion
    public Response updateDataset(@Auth AuthUser user, @Context UriInfo info, @PathParam("datasetId") Integer datasetId,  String json) {
```